config/config.go

```diff
@@ -7,24 +7,62 @@ import (
 	"net/url"
 	"os"
 
+	"strings"
+
 	"github.com/allegro/akubra/log"
 	"github.com/allegro/akubra/metrics"
 	set "github.com/deckarep/golang-set"
-	"github.com/go-yaml/yaml"
+	"github.com/go-validator/validator"
+	yaml "gopkg.in/yaml.v2"
 )
 
+// ClusterConfig defines cluster configuration
+type ClusterConfig struct {
+	// Backends should contain s3 backend urls
+	Backends []YAMLUrl `yaml:"Backends,omitempty"`
+	// Type, currently replicator is only option
+	Type string `yaml:"Type,omitempty"`
+	// Points how much load cluster should handle
+	Weight int `yaml:"Weight,omitempty"`
+	// Cluster type specific options
+	Options map[string]string `yaml:"Options,omitempty"`
+}
+
+// ClientConfig keeps information about client setup
+type ClientConfig struct {
+	// Client name
+	Name string `yaml:"Name,omitempty" validate:"regexp=^([a-zA-Z0-9_-]+)$"`
+	// List of clusters name
+	Clusters []string `yaml:"Clusters,omitempty" validate:"NoEmptyValuesSlice=Clusters,UniqueValuesSlice=Clusters"`
+}
+
+// YAMLUrl type fields in yaml configuration will parse urls
+type YAMLUrl struct {
+	*url.URL
+}
+
+// SyncLogMethod type fields in yaml configuration will parse list of HTTP methods
+type SyncLogMethod struct {
+	method string
+}
+
+// AdditionalHeaders type fields in yaml configuration will parse list of special headers
+type AdditionalHeaders map[string]string
+
 // YamlConfig contains configuration fields of config file
 type YamlConfig struct {
-	// Listen interface and port e.g. "0:8000", "localhost:9090", ":80"
-	Listen string `yaml:"Listen,omitempty"`
+	// Listen interface and port e.g. "0.0.0.0:8000", "127.0.0.1:9090", ":80"
+	Listen string `yaml:"Listen,omitempty" validate:"regexp=^(([0-9]+[.][0-9]+[.][0-9]+[.][0-9]+)?[:][0-9]+)$"`
+	// List of backend URI's e.g. "http://s3.mydatacenter.org"
+	Backends []YAMLUrl `yaml:"Backends,omitempty,flow"`
 	// Maximum accepted body size
-	BodyMaxSize string `yaml:"BodyMaxSize,omitempty"`
+	BodyMaxSize string `yaml:"BodyMaxSize,omitempty" validate:"regexp=^([1-9][0-9]+[kMG][B])$"`
 	// MaxIdleConns see: https://golang.org/pkg/net/http/#Transport
 	// Default 0 (no limit)
-	MaxIdleConns int `yaml:"MaxIdleConns"`
+	MaxIdleConns int `yaml:"MaxIdleConns" validate:"min=0"`
 	// MaxIdleConnsPerHost see: https://golang.org/pkg/net/http/#Transport
 	// Default 100
-	MaxIdleConnsPerHost int `yaml:"MaxIdleConnsPerHost"`
+	MaxIdleConnsPerHost int `yaml:"MaxIdleConnsPerHost" validate:"min=0"`
 	// IdleConnTimeout see: https://golang.org/pkg/net/http/#Transport
 	// Default 0 (no limit)
 	IdleConnTimeout metrics.Interval `yaml:"IdleConnTimeout"`
```

Comment:
We do it here: [metrics/config.go](https://github.com/allegro/akubra/blob/sharding/metrics/config.go#L19)